mesonbuild/modules/gnome.py

```diff
@@ -1477,7 +1472,8 @@ def mkenums_simple(self, state, args, kwargs):
             fhead += '%s\n' % body_prefix
         fhead += '#include "%s"\n' % hdr_filename
         for hdr in sources:
-            fhead += '#include "%s"\n' % os.path.basename(str(hdr))
+            relpath = hdr.rel_to_builddir(os.path.join(os.path.relpath('.', state.subdir), state.build_to_src))
```

Comment:
This does not work `hdr.rel_to_builddir(state.build_to_src)`? Everywhere else `rel_to_builddir` is used that's what is done.

Reply:
That produces a path relative to the build _root_. There's no guarantee the produced `enums` source will be in the root, and in fact, the `mkenums` test case does not have it there.

Reply:
Oh, I see, wouldn't be more clear as `os.path.join(hdr.rel_to_builddir(state.build_to_src), state.subdir)`? Or that's not the same?